evdns.c

```diff
@@ -5404,6 +5540,9 @@ evdns_getaddrinfo_gotresolve(int result, char type, int count,
 			/* If we have an answer waiting, and we weren't
 			 * canceled, ignore this error. */
 			add_cname_to_reply(data, data->pending_result);
+			if (!data->evdns_base->disable_cache) {
+				evdns_cache_write(data->evdns_base, data->nodename, data->pending_result, data->pending_result_ttl);
```

Comment:
`data->evdns_base` is `NULL` here - https://github.com/libevent/libevent/blob/970a258bc6ee08abf4cf81540570f0b8eca79449/evdns.c#L5358